include/nbl/builtin/hlsl/math/functions.hlsl

```diff
@@ -192,6 +192,21 @@ typename cpp_compat_intrinsics_impl::mul_helper<Lhs, Rhs>::return_t applyChainRu
     return hlsl::mul(dFdG, dGdR);
 }
 
+// takes in normalized vectors
+template<typename T>
+vector<T, 3> polarToCartesian(vector<T, 2> theta_phi)
+{
+    return vector<T, 3>(hlsl::cos(theta_phi.y) * hlsl::cos(theta_phi.x),
+                        hlsl::sin(theta_phi.y) * hlsl::cos(theta_phi.x),
+                        hlsl::sin(theta_phi.x));
+}
+
+template<typename T>
+vector<T, 2> cartesianToPolar(vector<T, 3> coords)
+{
+    return vector<T, 2>(hlsl::acos(clamp<float>(coords.z, -1, 1)), hlsl::atan2(coords.y, coords.x));
```

Comment:
should be `asin` for the `coord.z` cause you get it from a sin in `polarToCartesian`.

Also there should be no need to clamp, assert that input is already clamped

btw since we're in HLSL land, we should make a `struct polar<T> {T theta; T phil};` with specializations for `_static_cast_helper` and a header for it for semantic clarity and not being suspect to accidental casts betweeen cartesian and polar.

P.S. what uses it so far ?

Reply:
Used to bucket generated sample angles in the bxdf test